### PR TITLE
[Testcase Refactoring] Add proper hw_classification to test_core

### PR DESCRIPTION
**Pull request by TTTIAN998**
## Summary
- Add hw_classification annotations to test classes and aligning test methods with their hardware classification.

## Changes
- Add "GENERIC" hw_classification to TorchTensorTest and TorchTensorToFileTest in test/onnx/exporter/test_core.py.
- Import HardwareClassification from torch.testing._internal.common_utils.

## Test Plan
- python test/onnx/exporter/test_core.py.

## Notes
- This is part of the test case refactoring initiative tracked in [Test] PyTorch Test Case Refactoring and Track #185590.

### CODE REVIEW

File: test/onnx/exporter/test_core.py [hunk 1]
```diff
@@ -13,10 +13,12 @@
 import torch
 from torch.onnx._internal.exporter import _core
 from torch.testing._internal import common_utils
+from torch.testing._internal.common_utils import HardwareClassification
 
 
 @common_utils.instantiate_parametrized_tests
 class TorchTensorTest(common_utils.TestCase):
+    hw_classification = HardwareClassification.GENERIC
     @common_utils.parametrize(
```
Comment (lizhihai137) [inline test/onnx/exporter/test_core.py:23]: Add a blank line between the label attribute and the function.

File: test/onnx/exporter/test_core.py [hunk 2]
```diff
@@ -87,6 +89,7 @@ def test_tobytes_float4(self):
 
 
 class TorchTensorToFileTest(common_utils.TestCase):
+    hw_classification = HardwareClassification.GENERIC
     def _roundtrip_file(self, tensor: _core.TorchTensor) -> bytes:
```
Comment (lizhihai137) [inline test/onnx/exporter/test_core.py:95]: ditto

File: test/onnx/exporter/test_core.py (hunk 1 above)
Comment (TTTIAN998) [reply test/onnx/exporter/test_core.py:23]: Fixed.

File: test/onnx/exporter/test_core.py (hunk 2 above)
Comment (TTTIAN998) [reply test/onnx/exporter/test_core.py:95]: Fixed.

**Review by lizhihai137 — APPROVED** [review]

**Review by orangeH25 — APPROVED** [review]

**Review by fffrog — APPROVED** [review]
SGTM
